client/a8c-for-agencies/sections/marketplace/billing-dragon-checkout/index.tsx

```diff
@@ -26,24 +29,97 @@ const debug = debugFactory( 'a4a:bd-checkout' );
 function BillingDragonCheckoutContent( {
 	cartItems,
 	withA8cLogo = true,
+	siteSlug,
+	planSlug,
 }: {
 	cartItems: ShoppingCartItem[];
 	withA8cLogo?: boolean;
+	siteSlug?: string;
+	planSlug?: string;
 } ) {
 	const translate = useTranslate();
 	const [ isReady, setIsReady ] = useState( false );
 	const [ error, setError ] = useState< string | null >( null );
 
-	// Access the shopping cart API
-	const { replaceProductsInCart, responseCart } = useShoppingCart( 'no-site' );
-
+	const dispatch = useDispatch();
 	const agency = useSelector( getActiveAgency );
+	const site = useSelector( ( state ) => ( siteSlug ? getSite( state, siteSlug ) : undefined ) );
+	const siteId = site?.ID;
+	const isPlanCheckout = !! planSlug;
+
+	useEffect( () => {
+		if ( siteId ) {
+			dispatch( setSelectedSiteId( siteId ) );
+		}
+	}, [ dispatch, siteId ] );
+
+	// Use site's cart key when site exists, otherwise use 'no-site' for siteless checkout
+	const cartKey = siteId || 'no-site';
+	const { replaceProductsInCart, responseCart } = useShoppingCart( cartKey );
+
+	const {
+		productsForCart,
+		isLoading: areProductsPreparing,
+		error: productsError,
+	} = usePrepareProductsForCart( {
+		productAliasFromUrl: planSlug,
+		purchaseId: null,
+		usesJetpackProducts: false,
+		isPrivate: false,
+		siteSlug,
+		sitelessCheckoutType: 'a4a',
+	} );
 
 	debug( '[A4A Checkout] Cart items: ', cartItems );
 
+	useEffect( () => {
+		if ( ! isPlanCheckout || areProductsPreparing ) {
+			return;
+		}
+
+		if ( ! agency ) {
+			debug( '[A4A Checkout] Agency not loaded yet, waiting (plan checkout)...' );
+			return;
+		}
+
+		if ( ! productsForCart.length || productsError ) {
+			debug( '[A4A Checkout] No products created from plan slug' );
+			return;
+		}
+
+		const productsWithAgency = productsForCart.map( ( product ) => ( {
+			...product,
+			extra: {
```

Comment:
We'll need to set `isA4ASitelessCheckout: true` in the extra as well. That's used on the backend for a few thiings, including to set the meta to `is-a4a` which is important for recognizing an a4a checkout.
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Snqzva%2Qcyhtvaf%2Sjcpbz%2Qovyyvat%2Sfubccvat%2Qpneg.cuc%3Se%3Q7osnn8o8%231445-og

But there's also a check that tries to set the blog id since it assumes it's siteless - and has been up to this point. So we could change that to only happen when the blog id is actually empty (I think)
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Snqzva%2Qcyhtvaf%2Sjcpbz%2Qovyyvat%2Sfubccvat%2Qpneg.cuc%3Se%3Q7osnn8o8%231787-og

Or, we could pass some other parameter in `extra` and adjust the backend logic accordingly.